spring-cloud-function-web/src/main/java/org/springframework/cloud/function/web/function/FunctionEndpointInitializer.java

```diff
@@ -164,15 +167,19 @@ public void onApplicationEvent(ApplicationEvent event) {
 				ReactorHttpHandlerAdapter adapter = new ReactorHttpHandlerAdapter(handler);
 				HttpServer httpServer = HttpServer.create().host(address).port(port).handle(adapter);
 				Thread thread = new Thread(
-						() -> httpServer.bindUntilJavaShutdown(Duration.ofSeconds(60), this::callback),
+						() -> httpServer.bindUntilJavaShutdown(Duration.ofSeconds(60), (server) -> callback(server, context)),
 						"server-startup");
 				thread.setDaemon(false);
 				thread.start();
 			}
 		}
 
-		private void callback(DisposableServer server) {
+		private void callback(DisposableServer server, ApplicationContext context) {
 			logger.info("HTTP server started on port: " + server.port());
+			if (context instanceof ConfigurableApplicationContext) {
```

Comment:
This is actual a feature improvement as it will allow others to get at the local functional port (same as is available for Web and RSocket)